### ISSUE TITLE
Cypress cy.screenshot() timeout during `cypress run`

### ISSUE THREAD

**Issue by taylorpreston** [cypress-io/cypress#5016]
### Current behavior:
All of our tests pass locally, but when running the Codeship CI we get errors for a few tests.

The error during Codeship CI builds: `CypressError: cy.screenshot() timed out waiting '30000ms' to complete.`

### Desired behavior: 
All tests pass in Codeship CI

### Steps to reproduce: 

For our app the reproduction is running the `ci:codeship` command runs which calls `cy:record`
`
    "cy:record": "cypress run --record --key *key* --config video=false",
    "ci:codeship": "start-server-and-test start http://localhost:5000 cy:record"
`

I'm not exactly sure how to reproduce this other than creating a Codeship Basic app and attempting to run tests there. I have seen a few other issues point to a problem with how Cypress and Codeship interact. 

### Versions

Cypress version: 3.4.1

Sidenote: We are using the `cypress-autorecord` package, but it doesn't seem like this is involved.

**Comment by lleewwiiss** [comment]
@taylorpreston Did you ever resolve this issue? I am experience similar problems

**Comment by taylorpreston** [comment]
@lleewwiiss We never figured out what was causing this issue. We think it has to do with cypress beginning to run tests on the application before it has finished starting. The errors we received were the first 2-5 tests.

**Comment by zeel** [comment]
I am also experiencing the same error.

**Comment by justinpincar** [comment]
Also been getting this on 3.5.0 and 3.6.1.

**Comment by tibdex** [comment]
I also have this issue with 3.8.1 and Chrome 79 on CircleCI's `ubuntu-1604:201903-01` machine. It  wasn't there with Chrome 78.

**Comment by jennifer-shehane** [comment]
- Can you run Cypress in [debug mode](https://on.cypress.io/debugging#Debug-the-Command-Line) mode and print the entire set of logs here?
- Can you provide the machine and machine version they are running into this issue with, that would be appreciated. 
- Provide the browser you are running within (Electron, Chrome + version). 

I don't think this is related directly to https://github.com/cypress-io/cypress/issues/6023

**Comment by toneysum41** [comment]
@taylorpreston I meet the same issue, did you find out the solution for this issue?

**Comment by leichteckig** [comment]
Hello,
I have the same issue starting with 3.8.2 (and above) currently. Cypress 3.8.1 works correctly though.

We're using [cypress/included](https://github.com/cypress-io/cypress-docker-images/tree/master/included) docker image in Gitlab CI and keep getting these error messages:
```
CypressError: cy.screenshot() timed out waiting '60000ms' to complete.
```

It won't appear in every test and I can't see a pattern in those fails yet. At the moment, I'm struggling with getting debug mode to run, but I'll give an update as soon as I can provide some logs. I hope this first information can help nevertheless. Thank you in advance for all help, information etc.!

**Comment by bgcypressqa** [comment]
I can confirm with @leichteckig that this issue is occurring as described:

"... 3.8.2 (and above) currently. Cypress 3.8.1 works correctly though.

We're using cypress/included docker image in Gitlab CI and keep getting these error messages:

CypressError: cy.screenshot() timed out waiting '60000ms' to complete."

Only difference is that I am using Azure DevOps CI and their container services.

**Comment by teknologista** [comment]
I can alsoI confirm with @leichteckig that this issue is occurring as described.

We are using Gitlab CI with Ubuntu CI runners and your docker image cypress/browsers:node12.14.0-chrome79-ff71

CypressError: cy.screenshot() timed out waiting '60000ms' to complete."

random, masks errors also...It doesn't happen locally on desktop/laptop.

**Comment by Powersource** [comment]
I am having some variation on this issue. It was just happening for one specific test. Was running 3.6.1 and upgraded to 3.8.1, didn't help. When running the test locally I could see the real reason that the test was failing (instead of the screenshot timeout like CI was claiming). So for me it was mostly an issue of a bad error message i CI.

**Comment by teknologista** [comment]
@Powersource exactly what I meant by sometimes it also masks the true reason for the test fail

**Comment by alexbjorlig** [comment]
I just upgraded Cypress to `4.0.1`, and we now started to see tests failing on Github Actions with screenshot() timeout warning 😢

**Comment by jennifer-shehane** [comment]
### Need info

I still have questions that have gone unanswered in this thread that are necessary for investigation. 

- What browser are you running? What version of the browser?
- Are you running with `--headless` flag?
- What  machine and machine version are you running?
- Are you calling the `cy.screenshot()` command directly? Are you using a plugin that calls the `cy.screenshot()` command?

---

### Chrome 79 bug?

There has been some speculation that this may be caused by the Chrome 79 performance bug detailed here: https://github.com/cypress-io/cypress/issues/6023#issuecomment-584472171

If you are running on Chrome 79, we've released a docker image for Chrome 80 found here: https://github.com/cypress-io/cypress-docker-images/tree/master/browsers/node13.6.0-chrome80-ff72 at `cypress/browsers:node13.6.0-chrome80-ff72`.

Please try to update to using Chrome 80. If this is related to #6023 it may fix your issue.

**Comment by alexbjorlig** [comment]
Hi @jennifer-shehane.



> ### Need info
> I still have questions that have gone unanswered in this thread that are necessary for investigation.
> 
> * What browser are you running? What version of the browser? **Chrome 79.0.3945.130**
> * Are you running with `--headless` flag? **No**
> * What  machine and machine version are you running? **ubuntu-16.04**
> * Are you calling the `cy.screenshot()` command directly? Are you using a plugin that calls the `cy.screenshot()` command? **no**
> 
> ### Chrome 79 bug?
> There has been some speculation that this may be caused by the Chrome 79 performance bug detailed here: [#6023 (comment)](https://github.com/cypress-io/cypress/issues/6023#issuecomment-584472171)
> 
> If you are running on Chrome 79, we've released a docker image for Chrome 80 found here: https://github.com/cypress-io/cypress-docker-images/tree/master/browsers/node13.6.0-chrome80-ff72 at `cypress/browsers:node13.6.0-chrome80-ff72`.
> 
> Please try to update to using Chrome 80. If this is related to #6023 it may fix your issue.

I tried using chrome 80 in my Github actions:
```
jobs:
  e2eTests:
    runs-on: ubuntu-16.04
    container: cypress/browsers:node13.6.0-chrome80-ff72
```

Using the above container seems to give errors like this, and all tests fails:

```
[HPM] Error occurred while trying to proxy request /graphql from localhost:4200 to http://localhost:1337 (ECONNREFUSED) (https://nodejs.org/api/errors.html#errors_common_system_errors)
```

Our setup is an Angular + Graphql Node.js Express application. Is there something special I need to make this work when running in containers - or do you want me to open that as a separate issue?

But screenshots seems to be working 🔥

**Comment by jennifer-shehane** [comment]
@alexbjorlig Hmm, I haven't seen the ECONNREFUSED error for proxy requests before. Yeah, I would open a new issue. May be more due to the Node upgrade. 

The fact that screenshots work in Chrome 80 is very promising though. Thanks for the update.

**Comment by alexbjorlig** [comment]
@jennifer-shehane - thanks, I will create a new issue for the ECONNREFUSED.

**Comment by leichteckig** [comment]
Hello there,

A little update: I tried Cypress' docker images with Chrome 80 (Cypress 4.0.0 and newer) as recommended in https://github.com/cypress-io/cypress/issues/6023#issuecomment-584472171 and after a week or so, I don't see any more screenshot related failures until now. 

In 3.8.2, there was Chrome 79 installed iirc. So it looks like it was the mentioned Chrome bug for me as well. Thank you so much for your advice! :+1:

**Comment by jpike88** [comment]
Just upgraded to cypress/browsers:node12.16.1-chrome80-ff73 and this has broken my test builds using Codebuild.

**Comment by Undistraction** [comment]
We're seeing this on Chrome 80.0.3987.122 (Official Build) (64-bit)

**Comment by rhzs** [comment]
I have observed two different behaviors in **MacOS Mojave with Node JS 13 and Chrome v80.** 
1. Running directly `cypress run --browser chrome`, you will get a lot of timeouts. If it occurs once, then the next screenshot command will follow.
2. Running indirectly via `cypress open` then, select `chrome browser`. It successfully completes the tests.

This is weird!

**Comment by Legendofire** [comment]
> I have observed two different behaviors in **MacOS Mojave with Node JS 13 and Chrome v80.**
> 
> 1. Running directly `cypress run --browser chrome`, you will get a lot of timeouts. If it occurs once, then the next screenshot command will follow.
> 2. Running indirectly via `cypress open` then, select `chrome browser`. It successfully completes the tests.
> 
> This is weird!

We have had a similar issue, restarting our devices seemed to solve this particular issue so i assume it's device resources related.

**Comment by lleewwiiss** [comment]
@rhzs I believe this is due to some memory leak in caching test information. We have observed the same thing on machines with less ram (<=8GB. Setting `"numTestsKeptInMemory": 0` fixed it for us, but that makes debugging more difficult

**Comment by mingyixu** [comment]
We're also seeing this on Chrome 80.

**Comment by kamiloski** [comment]
I have seen it today too, it was first time

**Comment by shivasrini** [comment]
Seeing this when using the 4.2.0 included docker image. Tried reverting back to an older image 3.8.1 and I am still seeing it. It started happening when I tried to use log-to-console-output and I cant make it go away even when I have removed all references to it

**Comment by davojta** [comment]
I have the same problem with cypress 4.2.0 with electron browser running on TravisCI on ubuntu
I run locally in docker with 800mb and cant reproduce the issue.

**Comment by davojta** [comment]
Seems that problem is solved on Travis CI when I have start the cypress with `--headless --browser chrome` params. The chrome version is 73  in Travis CI.

**Comment by davojta** [comment]
Hm... It seems that setting numTestsKeptInMemory to 0 helps me and it is working with chrome 80 in Travis.

**Comment by saas2813** [comment]
I'm getting this problem on a Mac in the Electron browser.
I have a rather long page and the problem disappeared when I removed the cypress-failed-log plugin. I'm gonna have to try the other recommended plugin to work around the Issue "Capture + Display all logs for everything that happens in Cypress"
#448

**Comment by jennifer-shehane** [comment]
Still need some info on this one, as we've never encountered this error on any of our runs across CI. Some thoughts from others in this thread. 

- Setting "numTestsKeptInMemory": 0 may alleviate the problem
- Some plugins utilizing console.log output may be interfering with the screenshot somehow, like `cypress-failed-log` or `log-to-console-output`
- Restarting the device may help, as the device resources are an issue.

**Comment by rubin55** [comment]
@jennifer-shehane what are other consequences of setting numTestsKeptInMemory to 0? Is it still possible for cy.screenshot() to work when there are no tests kept in memory?

**Comment by jennifer-shehane** [comment]
1. `cy.screenshot()` will still work with `numTestsKeptInMemory` set to 0. Setting this to 0 makes it so that the application under test will not be 'restored' when you hover or click on a command as explained here: https://on.cypress.io/test-runner#Hovering-on-Commands
2. I don't know of any situation where this would occur or why this would interfere.

**Comment by JohnnyFun** [comment]
Fwiw, we started getting "cy.screenshot() timed out waiting 30000ms to complete." when we went from cypress@5.0.0 to 5.1.0. Reverting back to 5.0.0 seems to resolve so far.

We don't call `cy.screenshot` directly in any of our tests.

Running on:
- win10
- electron headless via `cypress run` command

I'll let you know if I see any patterns. Seems to happen on the same 3 or so tests, but not every run, and I don't think there's anything particularly special about them.

**Comment by amejiarosario** [comment]
Running into the same issue in our CI. It happens quite often. Using cypress@3.8.1 and CI is using Linux (CentOS).

Error:
`CypressError: cy.screenshot() timed out waiting '30000ms' to complete.`

Full stack trace:
```
  1) NetOps Report
       Validate charts
         should load charts for 90 day time range:
     CypressError: cy.screenshot() timed out waiting '30000ms' to complete.
      at Object.cypressErr (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:86089:11)
      at Object.throwErr (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:86044:18)
      at Object.throwErrByPath (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:86076:17)
      at https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:72522:21
      at tryCatcher (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:120353:23)
      at https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:115494:41
      at tryCatcher (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:120353:23)
      at Promise._settlePromiseFromHandler (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:118289:31)
      at Promise._settlePromise (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:118346:18)
      at Promise._settlePromise0 (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:118391:10)
      at Promise._settlePromises (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:118466:18)
      at Async../node_modules/bluebird/js/release/async.js.Async._drainQueue (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:115078:16)
      at Async../node_modules/bluebird/js/release/async.js.Async._drainQueues (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:115088:10)
      at Async.drainQueues (https://ci.dev.lockhart.io/__cypress/runner/cypress_runner.js:114962:14)
```

**Comment by rhzs** [comment]
Hi again Cypress team!

It seems, it is getting weirder and more unreliable than 6 months ago since I used the `cy.screenshot`. Now I often get the  `30000ms` timeout. 

Please help :'(.  It is one of the features what made me using Cypress.

**Comment by arvinderloyalty** [comment]
Hey, I am also facing this issue. Any updates?

**Comment by LyesAtt** [comment]
Hi people,

I came across the same problem, the screenshot was taking to much time and timed out my program resulting with an error.

Here's how I fixed it : **cy.screenshot('screenshot', clearTimeout(number));**

[](https://docs.cypress.io/guides/references/configuration.html#Timeouts
)

**Comment by bluepioupiou** [comment]
Same here, it seems to be a little chaotic : sometimes all goes well, and sometimes a few tests are failing with the 30000ms timeout. The screenshot stay stuck on the bottom of the page and timeout...

**Comment by davidwallacejackson** [comment]
@jennifer-shehane could you elaborate on how plugins using log output could cause this? I've started to see the issue as well, and I recently implemented a custom logging plugin. I'd be happy to instrument my test runs to help you collect data if you can tell me what'd be useful.

**Comment by jennifer-shehane** [comment]
@davidwallacejackson I'm not exactly sure why those would interfere, just that some other users indicated that those plugins may be causing the issue. We still need a clear reproducible way to reproduce this provided.

**Comment by arvinderloyalty** [comment]
For me, the reason why it was happening, there was a browser alert which was not handle and hence the test failed but cypress was not able to take the screenshot and hence it finally failed saying cy.screenshot() timeout

**Comment by jennifer-shehane** [comment]
@arvinderloyalty That makes sense. Like a normal Window alert?

**Comment by arvinderloyalty** [comment]
> @arvinderloyalty That makes sense. Like a normal Window alert?

@jennifer-shehane Yes, alert asking for credentials.

**Comment by harini910** [comment]
@jennifer-shehane I'm also facing the same issue CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.
I am running my cypress tests locally and here are the versions,

"cypress": "^5.2.0",
"mocha": "^5.2.0",
    "mochawesome": "^4.1.0",
    "mochawesome-merge": "^2.0.1",
    "mochawesome-report-generator": "^4.0.1",

I am trying to integrate screenshots to failed test cases in mochawesome report and as the screenshot is failing Iam unable to do it, please do help me on this and if there is any workaround please suggest.

**Comment by harini910** [comment]
> Hi people,
> 
> I came across the same problem, the screenshot was taking to much time and timed out my program resulting with an error.
> 
> Here's how I fixed it : **cy.screenshot('screenshot', clearTimeout(number));**

@LyesAtt - where did you add this line of code?

**Comment by LyesAtt** [comment]
In the spec.js file.

Is where I tell cypress everything it need to do.

But still sometimes, Cypress error screenshot timeout and get a failed. So the only thing is to ignore the error.

**Comment by harini910** [comment]
@LyesAtt - Thank you :) I will try and update.

**Comment by leonidesfernando** [comment]
> @jennifer-shehane I'm also facing the same issue CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.
> I am running my cypress tests locally and here are the versions,
> 
> "cypress": "^5.2.0",
> "mocha": "^5.2.0",
> "mochawesome": "^4.1.0",
> "mochawesome-merge": "^2.0.1",
> "mochawesome-report-generator": "^4.0.1",
> 
> running in chrome 86
> I am trying to integrate screenshots to failed test cases in mochawesome report and as the screenshot is failing Iam unable to do it, please do help me on this and if there is any workaround please suggest.

I'm getting the same issue  **CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete**.

Running my cypress tests locally and here are the versions,

    "mocha": "^8.2.0",
    "mochawesome": "^6.1.1",
    "mochawesome-merge": "^4.2.0",
    "mochawesome-report-generator": "^5.1.0",
    "cypress": ">=5.5.0"

And also trying to integrate screenshots to failed test cases in mochawesome report and as the screenshot is failing.

**Comment by BillyPapa** [comment]
Hi,

@jennifer-shehane  I am using Cypress with Chrome and only recently noticed (last few weeks) we keep getting the Error -

**CypressError: cy.screenshot() timed out waiting 30000ms to complete.**

Below is my package.json -
---------------------------------------------------------
{
  "name": "redress_e2e",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {
    "@angular/cli": "^8.3.29",
    "tsify": "^4.0.2",
    "typescript": "^3.9.7"
  },
  "devDependencies": {
    "@babel/core": "^7.11.6",
    "@babel/preset-env": "^7.11.5",
    "@cypress/webpack-preprocessor": "^5.4.6",
    "babel-loader": "^8.1.0",
    "cypress": "^3.8.0",
    "cypress-cucumber-preprocessor": "2.0.1",
    "cypress-iframe": "^1.0.1",
    "elliptic": "^6.5.3",
    "husky": "^3.1.0",
    "kind-of": "^6.0.3",
    "prettier": "^1.19.1",
    "pretty-quick": "^2.0.2",
    "ts-loader": "^8.0.4",
    "tslint": "^5.20.1",
    "tslint-config-prettier": "^1.18.0",
    "webpack": "^4.44.2"
  },
  "scripts": {
    "prettier:check": "prettier --check '**/*.{ts,js,css,json}'",
    "prettier:write": "prettier --write \"**/*.{ts,js,css,json}\""
  },
  "cypress-cucumber-preprocessor": {
    "nonGlobalStepDefinitions": true
  },
  "author": "",
  "license": "ISC",
  "husky": {
    "hooks": {
      "pre-commit": "pretty-quick --staged"
    }
  }
}
---------------------------------------------------------

The issue is intermittent but fails and gets the Error about 90% of the time.

**Chrome version:** 86.0.4240.183 (Official Build) (64-bit)

Is there an ETA when this Bug will be sorted?

**Comment by saas2813** [comment]
@jennifer-shehane When I run into this problem it was because the page that the screenshot was run on was rather long.
What caused me the most irritation was that the exception was thrown from inside the handling of a failed testcase, causing the test-case failure to be hidden. If there is an easy way to make sure that the test-failure is reported despite the screenshot (or anything else in the cleanup) failing, that would probably solve most of the irritation from this problem.

**Comment by karanshah-browserstack** [comment]
When we ran into this problem , we solved it by using a workaround by passing `"screenshotOnRunFailure": false` in your cypress.json file. In the scenario, the cypress doesn't try to capture the screenshot on failure and you get the actual error.

**Comment by sammacorpy** [comment]
Hi @jennifer-shehane facing this same issue consistently can we get an ETA when this will be fixed?

using modules 
```
  "cypress": "^5.6.0",
  "cypress-multi-reporters": "^1.4.0",
  "cypress-terminal-report": "^2.4.0",
  "mocha": "^8.2.1",
  "mochawesome": "^6.2.1",
  "mochawesome-merge": "^4.2.0",
  "mochawesome-report-generator": "^5.1.0",
```

browser: chrome Version 87.0.4280.88 (Official Build) (x86_64)
OS: macOS version 10.15.7

**Comment by jennifer-shehane** [comment]
We haven't been given a clear reproducible example of this issue. Without having a way to reproduce it, we can't track down when and why this error is throwing. There can't be any work done on the issue until that is provided. 

Can anyone provide a reproducible example?

**Comment by KrzysztofMadejski** [comment]
Had it as well when running locally in Docker

Cypress 6.2.1
Inside docker engine 20.10.0
on Darwin 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64

Electron 87 (headless) 

@jennifer-shehane The link to debugging command line you posted earlier "Can you run Cypress in debug mode mode and print the entire set of logs here?" do not work. I do not see any configuration option nor run parameter that enabled debugging.

**Comment by jennifer-shehane** [comment]
@KrzysztofMadejski https://on.cypress.io/troubleshooting#Print-DEBUG-logs

**Comment by RubenGarcia** [comment]
I am also affected. 
Cypress:    5.1.0                                                                              │
Browser:    Electron 83

**Comment by JohnnyFun** [comment]
@RubenGarcia you could try updating to latest 6.4.0?

**Comment by JosephHalter** [comment]
Version 6.4.0 works for me. Thank you.

**Comment by javiergarciavera** [comment]
it's happening the same for me with latest 6.4.0.
Cypress: 6.4.0
Browser: Chrome

**Comment by eselkin** [comment]
This might sound weird, but I experienced this ONLY when I had Norton antivirus firewall enabled.

**Comment by marracuene** [comment]
We get this in Chrome Headless on AWS Codebuild, with Cy 6.4.0 and 6.8.0.  Linking to [another Cypress issue](https://github.com/cypress-io/cypress/issues/9429) as for us, they seem to happen together.   N.B. the same tests do not fail when run locally.

We will try the fixes suggested by Jennifer above and update this comment.

**Comment by kaiyoma** [comment]
We also get this sometimes during our CI (Docker/k8s). We're running the latest version of Cypress (6.8.0) on Debian 9 Docker containers.

**Comment by AlexanderTunick** [comment]
Docker: cypress/base:12.16.2
Cypress versions: 6.5.0, 7.1.0
Where: in Jenkins

@jennifer-shehane come across this issue in Cypress version 6.5.0 and the latest 7.1.0. It gets reproduced only on a specific set of tests. There are ~7 tests. I've tested it at least 15 times running in Jenkins in Docker. Currently, it gets failed with _"CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete."_ on the current failure of the test : 

```
CypressError: `cy.task('eCommerceExecute')` failed with the following error:

> Response code 422 (Unprocessable Entity)

```


I run the tests as a node process using cypress as an imported lib as you all suggest in one of your docs. Following this, I did not manage to set the DEBUG parameter. Could you please recommend what else data I could extract for you to be useful to provide. Also if it appears not enough I'm ready to jump on a call to work on this case together since I can't just send this test as it involves many integrations which will be a bit tough to mock.

**Comment by mihaellenic** [comment]
Experiencing the same issue on Cypress 7.1.0 running inside `cypress/browsers:node12.18.3-chrome89-ff86` docker image.

Error message: `CypressError: 'cy.screenshot()' timed out waiting '30000ms' to complete.`

**Comment by jennifer-shehane** [comment]
@AlexanderTunick Can you provide a reproducible example?

**Comment by pietrofxq** [comment]
I am having the same problem, from my investigation this seem to be happening if another tab is opened during the test. This issue is happening on two of my tests that presents this behaviour: First is testing our Chrome Extension, that opens a welcome page after being installed. The second one was on a test that opens another tab on purpose, to test if our app can be used on a single tab only

**Comment by BuckyB-test** [comment]
My team is facing this problem in windows machine. @jennifer-shehane Can you try this out on a page which needs to be scrolled thrice to take a screenshot. (Can't provide the example here!!)

On such a page perform some action and try to take screenshot

cy.screenshot()
//some steps on top of page
cy.screenshot()
//some steps in top of page
cy.screenshot()
//some actions in middle of the page
cy.screenshot()
some action towards the end of the page

**Comment by talaikis** [comment]
+1, random timeouts on live site, works fine locally.

**Comment by abejfehr** [comment]
Is there a way to catch these failures to screenshot? It would be nice if those didn't result in a `CypressError` and didn't get treated as test failures

**Comment by bvellacott** [comment]
we are experiencing this on cypress 6.4.0

**Comment by lucascandidodias** [comment]
I have this problem too, however I noticed that the error only happens when I minimize the browser screen. If I run the tests with it maximized, I don't suffer from this problem. And if I run the tests in headless mode, I can also run them without any problem.

The version of cypress I am using is 6.6.0

**Comment by BillyPapa** [comment]
Hi,

Please remove me from this email trail.

Thanks

BP

Billy Papageorgiou
Developer
CRS Engineering, Customer Relationship Systems
CUSTOMER PROCESSING SYSTEMS
***@***.***
Level 5 South, 55 Currie Street, Adelaide

**Comment by divpusher** [comment]
**OS:** Windows 10.
**Cypress:** 8.7.0
**Browser:** Electron 93 (headless)

I tried **cypress-plugin-snapshots** and **cypress-image-snapshot** plugins, both had similar results.

If I have `video: false` setting in **cypress.json** file and run Cypress from command line, it throws this error: 

`CypressError: cy.screenshot() timed out waiting 30000ms to complete.`

However, it works when I run the test via Cypress UI.

**test.spec.js**
```
describe('Screenshot testing', function() {
  it('page exists and can be opened', function() {
    cy.visit('http://local.testurl.com:3001').then(() => {
      cy.document()
        .toMatchImageSnapshot();
    })
  })
})
```

**Comment by BillyPapa** [comment]
Hi,

Please remove me from this email list.

Thanks.

Billy

**Comment by divpusher** [comment]
Just click on the **unsubscribe** link in the bottom of the email...

**Comment by yoavniran** [comment]
I can confirm that when running with `video: false` when there are tests failing, cypress will time out ([running in circle-ci](https://app.circleci.com/pipelines/github/rpldy/react-uploady/993/workflows/36677187-ef38-4dfe-817c-d19f2d19d054/jobs/3353))
Its Circle that kills the process eventually rather than the process exiting properly. This makes the tests run very long of course needlessly.

When changing to `video: true` cypress will exit immediately after tests finish running.  (see [run](https://app.circleci.com/pipelines/github/rpldy/react-uploady/997/workflows/fc215d90-5ec0-40a7-aa3d-576814377ea3/jobs/3361) after change)

I have no desire for the videos to be captured but seems like I have to enable this feature until this is fixed.

**using cypress 9.0.0**

**Comment by Arunk28** [comment]
CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.

version :8.7.0

occurs frequently in all the suites
why this is open for years ? 

can anyone please take this as priority and fix it 
it will be helpful for many 

@jennifer-shehane

**Comment by AdrianoFerrari** [comment]
I recently updated from ^6.9.1 to ^9.1.0 and now I get this cy.screenshot timed out error on failed tests, and the process doesn't exit (Github Action ends up running for hours, if I don't manually cancel).

I also have `video: false` in my config.

**Comment by danatemple** [comment]
We have also now started to get this issue (including in trivial tests that do not actively take screenshots) having upgraded from 7.3 to 9.1.
Like @yoavniran and @AdrianoFerrari we also have **video: false**, and also see that some tests are not terminating. Not sure if related.
We'd like to keep the version current, but this is a blocker.

**Comment by testn** [comment]
I'm seeing the same behavior on 9.1.1 on Jenkins as well

**Comment by sramshek** [comment]
I am also seeing this behavior !

**Comment by adrianiskandar** [comment]
Seeing the same behavior in Cypress 8.7. The worst part is that the process doesn't end, it keeps hanging, so even if you try to catch the error you can't close the process.

**Comment by DianaPlesa92** [comment]
I am having the same issue on 9.0.0. Running my tests with video:false since I do not need the video and it is taking too long to run them with video. The run is in headless mode.

**Comment by filipedanunes** [comment]
happening the same on 9.2.0 running locally

**Comment by laravanwijk** [comment]
I am also seeing this for the first time after upgrading to a 9.0.0 or higher version. The issue does not show up locally, only in our Jenkins build. Setting video:true solved this, but this is not desirable as a long-term solution since we don't have a need for the videos.

**Comment by talaikis** [comment]
This didn't worked for me since v7. Switched to playwright, it produces much more stable tests, not just screenshots.

**Comment by vasiliy-ilto** [comment]
any progress on this? this issue open for years...

**Comment by trv-wcaneira** [comment]
To try and help, I created a minimal repro of the issue [here](https://github.com/trv-wcaneira/cy-screenshot-timeout) with some README details on the environment I'm using, and a snippet of the last line of the DEBUG logging I sometimes see before it "hangs"/times out.

I also noticed some interesting cause/effect observations in the README (some of which have been noted here by others).  For example, I notice that invoking `cy.screenshot()` directly doesn't fail for me, only when it's done indirectly by a failed test assertion.  When I _do_ invoke it directly, it seems to make subsequent failed assertions & screenshots _NOT_ timeout.

Mentioning @jennifer-shehane as I kindly saw her asking for these details above.  Hoping this will help.

Our workaround for now is to enable `video: true` which works (as stated by others) but is obviously not ideal if you don't want the video.

**Comment by brainfoolong** [comment]
I came across this issue as well with `9.3.1` and i already started to create a new issue, but finally found this one. Exact same problem here with `video:false`, it hangs an never exit on failed tests. `video:true` is no option for us, as we have automated headless tests on not so beefy machines.

**Comment by trv-wcaneira** [comment]
> I notice that invoking `cy.screenshot()` directly doesn't fail for me, only when it's done indirectly by a failed test assertion. When I _do_ invoke it directly, it seems to make subsequent failed assertions & screenshots _NOT_ timeout.

> Our workaround for now is to enable `video: true` which works (as stated by others) but is obviously not ideal if you don't want the video.

Based on the observation above, and just for fun, I tried reverting `video: false` and putting an explicit `cy.screenshot()` call before my tests e.g.

```js
//in support/index.js

before(() => {
  cy.screenshot()
})
```

...but it didn't work consistently, yielding an occasional timeout and even an error code 3221225477.  

At the moment I'm sticking with `video: true`.

**Comment by danatemple** [comment]
@jennifer-shehane this is getting more and more problematic. Any change you can get this on the fix list?

I just timed running with `video: true`, it was generally **3-4 x slower** than without (!) and this is on beefy c5.large AWS instances.

This issue appeared for us when attempting to upgrade to Cypress 8.x - we are stuck on 7.x as a consequence.

**Comment by thoover11** [comment]
A workaround that worked for me:

I found that this was occurring for me if during each describe block, my first action was to do a `cy.visit(<url>)`, then running tests that would fail with NO screenshot.

After playing around with what was semi working, I tried assigning the url I wanted to visit at the beginning of each test to the baseUrl: <url> configuration in either cypress.json or plugins/index.js module.exports config.  Then instead of using cy.visit(<url>), I change my in-test visit() to `cy.visit('/').  Doing this caused my screenshots to work while having videos disabled/set to false.

**Before:**
```
cypress.json
{
  ...,
  "video": false
}

dummy.spec.js:
describe('should fail', () => {
    it('visit url', () => {
         cy.visit('localhost:3000')
    }

    it('test that should've failed and triggered a screenshot', () => {
         cy.get('[data-testid=header]').contains(<incorrect title>) 
    }
}
```

**After:**
cypress.json
{
  ...,
  "video": false,
  "baseUrl": <url>
}

dummy.spec.js:
`describe('should fail', () => {
    it('visit url', () => {
         cy.visit('/')
    }

    it('test that failed and SUCCESSFULLY trigger a screenshot', () => {
         cy.get('[data-testid=header]').contains(<incorrect title>) 
    }
}`

**Comment by QAcodes** [comment]
Hi Everyone,
For those Experiencing this issue on Cypress version ^9.4.1 , I just resolve this issue by updating the **Node.js Version** to 16 or to the latest version (https://nodejs.org/en/ ) in our setup. 

Hopefully it could help Thanks!

**Comment by trv-wcaneira** [comment]
> After playing around with what was semi working, I tried assigning the url I wanted to visit at the beginning of each test to the baseUrl: configuration in either cypress.json or plugins/index.js module.exports config. Then instead of using cy.visit(), I change my in-test visit() to `cy.visit('/'). Doing this caused my screenshots to work while having videos disabled/set to false.

Recently our CI runs grew slower, and running with `video: true` seemed not to help, so instead I tried this `baseUrl` solution (via `config` CLI flag, in our case) and _so far_ it seems to work cleanly, not requiring video (`video: false`), and still producing expected screenshots and helpful error messages on test failures.  We're on version 9.2.1 still.

We had been setting `baseUrl` at the start of each test, where setting `baseUrl` up front (CLI, cypress.json, etc) is really a documented Cypress best practice we probably should have been doing anyway.

Time will tell if we have more issues, but for now, I just wanted to share in case this helps anyone else.  Thank you, @thoover11.

**Comment by cavb** [comment]
What worked for me on version `9.5.0`  was disabling video and screenshots on test failure on the `cypress.json` config:

```javascript

{
    ...
    "video": false,
    "screenshotOnRunFailure": false
}
```

**Comment by marracuene** [comment]
> What worked for me on version `9.5.0` was disabling video and screenshots on test failure on the `cypress.json` config:
> 
> ```js
> {
>     ...
>     "video": false,
>     "screenshotOnRunFailure": false
> }
> ```

The challenge is.. when a test fails is often when you most need to see the video, if you are running in a headless build environment and need to _ex post facto_ work out why a certain test failed.

**Comment by cavb** [comment]
> The challenge is.. when a test fails is often when you most need to see the video, if you are running in a headless build environment and need to _ex post facto_ work out why a certain test failed.

I agree 100%. This should be fixed by Cypress.

But this worked for my use case, I wanted to disable for local runs so my CPU doesn't get too hot :fire:

**Comment by LuigiMaestrelli** [comment]
Any news on this?

**Comment by mycarrysun** [comment]
> A workaround that worked for me (using v9.3.1):
> 
> I found that this was occurring for me if during each describe block, my first action was to do a `cy.visit(<url>)`, then running tests that would fail with NO screenshot.
> 
> After playing around with what was semi working, I tried assigning the url I wanted to visit at the beginning of each test to the baseUrl: configuration in either cypress.json or plugins/index.js module.exports config. Then instead of using cy.visit(), I change my in-test visit() to `cy.visit('/'). Doing this caused my screenshots to work while having videos disabled/set to false.
> 
> **Before:**
> 
> ```
> cypress.json
> {
>   ...,
>   "video": false
> }
> 
> dummy.spec.js:
> describe('should fail', () => {
>     it('visit url', () => {
>          cy.visit(<url>)
>     }
> 
>     it('test that should've failed and triggered a screenshot', () => {
>          cy.get('[data-testid=header]').contains(<incorrect title>) 
>     }
> }
> ```
> 
> **After:**
> 
> ```
> cypress.json
> {
>   ...,
>   "video": false,
>   "baseUrl": <url>
> }
> 
> dummy.spec.js:
> describe('should fail', () => {
>     it('visit url', () => {
>          cy.visit('/')
>     }
> 
>     it('test that failed and SUCCESSFULLY trigger a screenshot', () => {
>          cy.get('[data-testid=header]').contains(<incorrect title>) 
>     }
> }
> ```

This didn't work for me unfortunately (Cypress v9.5.3, ubuntu 20.04, circle ci) - everything works locally just fine but not in CI

**Comment by Arunk28** [comment]
> Any news on this?

try 8.5 it wont occur

**Comment by adnansarajlic** [comment]
This needs to be addressed @jennifer-shehane

**Comment by glassblowerscat** [comment]
For anyone who can't figure out another workaround besides `video: true`, here's what I've done to keep the compression process from eating up all my CI/CD time and resources:

In `cypress/plugins/index.js`:

```
module.exports = (on, config) => {
  on("after:spec", (_, results) => {
    return fs.unlinkSync(results.video);
  });
  on("after:run", () => {
    fs.rmdirSync(`cypress/videos`, { recursive: true });
  });

  return config;
};
```

The `after:spec` hook is what actually makes the difference; it seems to delete the video file before the compression process can start. `after:run` is just for tidiness 😁

**Comment by marracuene** [comment]
> For anyone who can't figure out another workaround besides `video: true`, here's what I've done to keep the compression process from eating up all my CI/CD time and resources:
> 
> In `cypress/plugins/index.js`:
> 
> ```
> module.exports = (on, config) => {
>   on("after:spec", (_, results) => {
>     return fs.unlinkSync(results.video);
>   });
>   on("after:run", () => {
>     fs.rmdirSync(`cypress/videos`, { recursive: true });
>   });
> 
>   return config;
> };
> ```
>
> The `after:spec` hook is what actually makes the difference; it seems to delete the video file before the compression process can start. `after:run` is just for tidiness grin

Worth a try, thankyou.   What version of Cy are you using?

**Comment by glassblowerscat** [comment]
> Worth a try, thankyou. What version of Cy are you using?

This is on 9.5.4. It does seem like this takes a bit longer than with video completely off, but it's hard to do apples-to-apples comparisons. Definitely it takes less time than with video compression running

**Comment by rkhomi** [comment]
Same on 9.5.0

**Comment by ArktinenSieni** [comment]
Any progress in the matter?

**Comment by qeviking** [comment]
Still happening on 9.7

**Comment by dante01yoon** [comment]
it also happens on 9.5.0. 
by the way, how hilarious is this solution is go back to version 7.7. 
It's been only couple of days cypress v10 has come.

**Comment by dlpigpen** [comment]
I got the same error with Chrome 1.0.2 Cypress 9.7
```
Cypress:        9.7.0                                                                          │
  │Browser:        Chrome 102                                                                     │
  │Node Version:v16.15.1 (/opt/hostedtoolcache/node/16.15.1/x64/bin/node)│
  ```

```
       should have the correct title and powered by:
     CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.
      at http://localhost:4200/__cypress/runner/cypress_runner.js:165578:81
      at tryCatcher (http://localhost:4200/__cypress/runner/cypress_runner.js:13022:23
      ```

**Comment by dante01yoon** [comment]
I think just set `"screenshotOnRunFailure": false` in CI would be quiet reasonable work around, of course it's not a best solution if you want to log where error occurs.

**Comment by ArktinenSieni** [comment]
Besides the screenshots themselves failing, I find it worrying that the timeout does't throw an error in CI environment thus tests running indefinetly. At least when using GH Actions.

**Comment by dlpigpen** [comment]
I fixed this issue by updating all error tasks in my testcase. But until now i still don't know a reason why.

**Comment by glassblowerscat** [comment]
> > Besides the screenshots themselves failing, I find it worrying that the timeout does't throw an error in CI environment thus tests running indefinetly. At least when using GH Actions.
> 
> It throws an error for us in GH actions - I bet there's something else causing your issue

I see this behavior intermittently. Maybe one out of every 20 test runs, an error doesn't throw and the test just hangs indefinitely. Haven't figured out a fix, except that I'm planning to put a timer on my CI/CD runs

**Comment by romankhomitskyi** [comment]
Could you guys run cypress in debug mode to see why cy.screenshoot() fails?

**Comment by dlpigpen** [comment]
> Could you guys run cypress in debug mode to see why cy.screenshoot() fails?

I actually don't know why. To fix it i turned off capturing screenshort or make the testcase works.

**Comment by romankhomitskyi** [comment]
> > Could you guys run cypress in debug mode to see why cy.screenshoot() fails?
> 
> I actually don't know why. To fix it i turned off capturing screenshort or make the testcase works.

That is obvious 😁

**Comment by danatemple** [comment]
@romankhomitskyi sorry for the delay, I've now had time to run both a passing and failing run (that hangs at the end) with debug on.

[cypressDebugHang.log](https://github.com/cypress-io/cypress/files/9187697/cypressDebugHang.log)
[cypressDebugOk.log](https://github.com/cypress-io/cypress/files/9187698/cypressDebugOk.log)
[cypressDebugOkScreenshotFalse.log](https://github.com/cypress-io/cypress/files/9187882/cypressDebugOkScreenshotFalse.log)
[cypressDebugOkVideoFalse.log](https://github.com/cypress-io/cypress/files/9187884/cypressDebugOkVideoFalse.log)

This is a run of 10 tests, where one of these tests fails. There are two basic log files:

```
cypressDebugHang.log: Test fails, Cypress hangs indefinitely at the end of the run.
cypressDebugOk.log: If the failing test is skipped, then Cypress exits at the end.

```
The above are with video: false (as this slows all tests) and screenshots not disabled (screenshotOnRunFailure: true) for CI usage.

Also, if we set `video: true`, or disable screenshots, the hang does not occur. These logs are in `cypressDebugOkVideoFalse.log` and `cypressDebugOkScreenshotFalse.log`.

Hope this helps, let me know if you want further debug.

**Comment by alexsch01** [comment]
Does anyone have an update on this issue? It's about to be 3 years and no fix

**Comment by darkmasta** [comment]
Problem persists.

**Comment by Arunk28** [comment]
> t to be 3 years and

@jennifer-shehane

**Comment by romankhomitskyi** [comment]
It might be that your CI machine does not have enough resources to run your tests

**Comment by LRejkowski** [comment]
Hi I have Cypress version 10.1.0 and I got:

screenshot
CypressError
cy.screenshot() timed out waiting 50000ms to complete.


Stack trace
 Print to console
    at <unknown> (https://page/__cypress/runner/cypress_runner.js:158612:82)
From previous event:
    at automateScreenshot (https://page/__cypress/runner/cypress_runner.js:158611:12)
    at scrollAndTake (https://page/__cypress/runner/cypress_runner.js:158685:13)
From previous event:
    at takeScrollingScreenshots (https://page/__cypress/runner/cypress_runner.js:158688:59)
    at takeFullPageScreenshot (https://page/__cypress/__cypress/runner/cypress_runner.js:158722:11)
    at <unknown> (https://page/__cypress/__cypress/runner/cypress_runner.js:158954:15)
From previous event:
    at takeScreenshot (https://page/__cypress__cypress/runner/cypress_runner.js:158942:23)
    at Context.screenshot (https://page/__cypress/__cypress/runner/cypress_runner.js:159062:15)
    at Context.<anonymous> (https://page/__cypress/__cypress/runner/cypress_runner.js:171662:20)
From Your Spec Code:
    at Object.followLink (webpack:///./cypress/page-objects/pageObjectName.ts:33:5)
    at Context.eval (webpack:///./cypress/e2e/System_Tests/testName.cy.ts:80:19)

**Comment by CrisDgrnu** [comment]
I'm still getting this error

**Comment by preetibellad** [comment]
I am getting same error with Cypress version 10.11.0:
CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.
with Webkit run.

Stacktrace:
     CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.

Because this error occurred during a `after each` hook we are skipping the remaining tests in the current suite: `User should be able to chec...`
  tryCatcher@http://localhost:3000/__cypress/runner/cypress_runner.js:11318:28
  tryCatcher@http://localhost:3000/__cypress/runner/cypress_runner.js:11318:28
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:9253:35
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:9310:43
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:9355:24
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:9431:33
  _drainQueueStep@http://localhost:3000/__cypress/runner/cypress_runner.js:6025:27
  _drainQueue@http://localhost:3000/__cypress/runner/cypress_runner.js:6018:24
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:6034:16
  <unknown>@http://localhost:3000/__cypress/runner/cypress_runner.js:5904:26

**Comment by danatemple** [comment]
@romankhomitskyi @jennifer-shehane We are still stuck on Cypress 7 due to this issue appearing. It's a total blocker as any test run risks hitting this and then hanging.
Any reason to believe it might have been fixed in version 10+?
I provided a bunch of debug logs as requested above. Willing to assist further debug.

**Comment by romankhomitskyi** [comment]
> @romankhomitskyi @jennifer-shehane We are still stuck on Cypress 7 due to this issue appearing. It's a total blocker as any test run risks hitting this and then hanging. Any reason to believe it might have been fixed in version 10+? I provided a bunch of debug logs as requested above. Willing to assist further debug.

Stll happening on 12.5.0

But only on our servers which appear to be not powerful 

When using aws spot instance - no issues

**Comment by anjalichaudhri** [comment]
the issue is also occurring in the embedded electron browser.  cypress version: 12.3.0

**Comment by rooby** [comment]
This is happening for me also. I am not calling cy.screenshot() directly, it is happening when Cypress tries to take a screenshot on test failure. Screenshots do work some times, but they fail at least half the time there is an error to take a screenshot of it seems.

- Cypress: 9.7.0
- Environment: docker.io/cypress/included:9.7.0 (nothing overridden)
- Platform: AWS CodeBuild, 15GB RAM, 8 vCPU
- Browser: Chrome (headless)

Using config
```
{
  "baseUrl": "http://localhost:4200",
  "retries": {
    "openMode": 0,
    "runMode": 2
  },
  "screenshotOnRunFailure": true,
  "video": false,
  "viewportHeight": 1080,
  "viewportWidth": 1920
}
```
Debug log: [cypress-screenshot-bug-debug-log.txt](https://github.com/cypress-io/cypress/files/11209231/cypress-screenshot-bug-debug-log.txt)

Workarounds mentioned in this issue:

- Add base URL and visit / on each test: I already do this, visiting / in before each.
- Turn on video: This didn't fix it for me. It is also not an acceptable workaround for us as we don't need video and we can't take the run time increase.
- Turn off screenshot on failure: Haven't tried. This is not an option as the thing we're troubleshooting is the screenshot feature.
- Set numTestsKeptInMemory = 0: This did not fix the issue for me.

**Comment by jordanpowell88** [comment]
@rooby  Have you tried upgrading to Cypress 12? There are significant performance improvements that might resolve your issue

**Comment by mycarrysun** [comment]
@jordanpowell88 https://github.com/cypress-io/cypress/issues/5016#issuecomment-1488008758 - still happening in 12.3.0

**Comment by rooby** [comment]
> @rooby Have you tried upgrading to Cypress 12? There are significant performance improvements that might resolve your issue

I haven't tried yet. We have our upgrade to 12 planned, which will hopefully happen soon.
I'll give it another investigation once that's done, but based on other comments in this thread I expect maybe it won't change anything.

**Comment by jordanpowell88** [comment]
> @jordanpowell88 [#5016 (comment)](https://github.com/cypress-io/cypress/issues/5016#issuecomment-1488008758) - still happening in 12.3.0

There has been obviously, quite a bit of work that has gone on between current Cypress 12+ and version 7. I would love to see if upgrading resolved it in your instance. It seams like performance is the potential culprit based upon some comments in the thread and performance has improved quite a bit in that span.

**Comment by jordanpowell88** [comment]
Can anyone provide me some more context or a reproducible example? It's very difficult to investigate an issue that appears to be inconsistant.

**Comment by jordanpowell88** [comment]
> I haven't tried yet. We have our upgrade to 12 planned, which will hopefully happen soon.
> I'll give it another investigation once that's done, but based on other comments in this thread I expect maybe it won't change anything.

No rush. I hope that resolves your issue. Does it happen in CI or local?

**Comment by romankhomitskyi** [comment]
> > I haven't tried yet. We have our upgrade to 12 planned, which will hopefully happen soon.
> > I'll give it another investigation once that's done, but based on other comments in this thread I expect maybe it won't change anything.
> 
> No rush. I hope that resolves your issue. Does it happen in CI or local?

This is still happening on 12.5.0, only on CI machine which does not have enough resources I guess

**Comment by leenattress** [comment]
I want to leave a comment here, because we found the cause of this issue and its something we could work around:

Our tests had a link that when clicked opened a new tab. The tab was on another domain and the browser could not perform `cy.screenshot()`.

We changed the test so that it asserted this another way and it passed through fine.

We were using [Lambdatest](https://www.lambdatest.com) when this problem manifested itself.

- Check your domains
- Check for new tabs
- They are fast, faster than you could believe. Don't turn your back, don't look away, and *don't blink*!

**Comment by tchaton** [comment]
Any updates ?

**Comment by theTestingApproach** [comment]
Happening in 12.16.0 as well.
Chrome 114
Any other info required please do let know please.
thanks

**Comment by theTestingApproach** [comment]
@jennifer-shehane , hi
is no one checking this issue please? i am stuck to using version 12:12.0 and would really like to update the version.
Please do let me know if any other info is required.
Thank you

**Comment by mycarrysun** [comment]
@theTestingApproach don't get your hopes up - this issue has been stale from the core cypress team for years

**Comment by wongk** [comment]
I am running into this issue running a specific script in headless on my local.

**Comment by marracuene** [comment]
> @jennifer-shehane , hi is no one checking this issue please? i am stuck to using version 12:12.0 and would really like to update the version. Please do let me know if any other info is required. Thank you

Be careful... we upgraded and ran in to this: https://github.com/cypress-io/cypress/issues/23903

Really frustrated with Cypress ATM ... spending a lot of cycles dealing with framework issues rather than actually writing/reasoning about tests.

**Comment by theTestingApproach** [comment]
Honestly what the point of releasing new versions if people are not using it.

**Comment by fzampori-sumo** [comment]
+1

**Comment by Arunk28** [comment]
cy.screenshot is major error...  why you guys not fixing for years .. ? 
@jennifer-shehane  ? ? ? ?

**Comment by ghost** [comment]
Just an observation: 
Within one of our Cypress projects, this seems to occur most often when the web page is stuck in a loading state. 

However when Cypress goes to take a screenshot of the underlying failure, it is unable to perform the action. This could be related to high cpu/memory usage of the loading page. 

When reading the results instead of stating the underlying failure, it reports "`cy.screenshot()` timed out waiting `30000ms` to complete"  overriding the underlying failure. 

I was only able to notice this as I checked the test body in my Allure report and even there the actually failure was hidden.

**Comment by 403-html** [comment]
I had the same error. In my case it turned out that the some test failed earlier, but I guess when connecting to the dashboard it didn't send the information in time to fail the whole run and went on further. It was stuck on the "good" test in `beforeEach`, however, the failed test that should have fail the whole run was earlier.

```
Cypress: 12.17.0
Browser: Electron 106 (headless) 
Node Version: v18.10.0 (/home/runner/.nvm/versions/node/v18.10.0/bin/node)
```

**Comment by fzampori-sumo** [comment]
I have same issue with cypress@12.17.3 and chrome

**Comment by theTestingApproach** [comment]
I do not understand how such a critical issue is not being addressed..

**Comment by marktnoonan** [comment]
Hi all, I first want to say I can understand the frustration with an issue like this. Especially when it seems simple, like there is a single solution. But from what I can tell, it's not just one error, but it's a condition Cypress reaches that could point back to multiple causes. That's one of the challenges with just providing a fix.

Anyway, a small update: it appears that Chrome's new headless mode that was turned on in [v12.15.0](https://docs.cypress.io/guides/references/changelog#12-15-0) is connected to at least some the more recent reports of problems above. That's suggested in this Discord [thread from today](https://discord.com/channels/755913899261296641/1138563361085333596/1138563361085333596), as well as the increased activity in these comments after 12.15.0 came out in June. 

You can still opt-in to passing the old headless flag, as [described in our docs here](https://docs.cypress.io/api/plugins/browser-launch-api#Disable---headlessnew-for-Chrome), which should get you back to the default behavior prior to v12.15.0. If you started seeing this issue recently, and this is the reason, that should unblock you while we look into it further.

Also I am really curious if setting [`experimentalMemoryManagement`](https://docs.cypress.io/guides/references/experiments) to `true` in your config file might also make difference here. I may have missed it above, if anybody who experiences it regularly has already configured that, or if not, please let me know if changes the amount of times this happens.

**Comment by SaurabhBose** [comment]
> Hi all, I first want to say I can understand the frustration with an issue like this. Especially when it seems simple, like there is a single solution. But from what I can tell, it's not just one error, but it's a condition Cypress reaches that could point back to multiple causes. That's one of the challenges with just providing a fix.
> 
> Anyway, a small update: it appears that Chrome's new headless mode that was turned on in [v12.15.0](https://docs.cypress.io/guides/references/changelog#12-15-0) is connected to at least some the more recent reports of problems above. That's suggested in this Discord [thread from today](https://discord.com/channels/755913899261296641/1138563361085333596/1138563361085333596), as well as the increased activity in these comments after 12.15.0 came out in June.
> 
> You can still opt-in to passing the old headless flag, as [described in our docs here](https://docs.cypress.io/api/plugins/browser-launch-api#Disable---headlessnew-for-Chrome), which should get you back to the default behavior prior to v12.15.0. If you started seeing this issue recently, and this is the reason, that should unblock you while we look into it further.
> 
> Also I am really curious if setting [`experimentalMemoryManagement`](https://docs.cypress.io/guides/references/experiments) to `true` in your config file might also make difference here. I may have missed it above, if anybody who experiences it regularly has already configured that, or if not, please let me know if changes the amount of times this happens.

Hello @marktnoonan,
With v12.17.2, I just tried the `experimentalMemoryManagement` and have not faced any timeout failures since. The running time also seemed similar as before so no regression there. Will update in case face any issues. But for now this seems to be a good solution for us. 👍

**Comment by marktnoonan** [comment]
Glad to hear this @SaurabhBose 

If the `cy.screenshot()` timeout is a symptom of an out-of-memory error, the `experimentalMemoryManagement` would help. I'd like to hear from a few other people before we call this a true workaround. Also glad perf was not impacted for you.

**Comment by kuldea-paulo** [comment]
Hi @marktnoonan. 
Unfortunately, In my case, `experimentalMemoryManagement: true` didn't help. I still have the `cy.screenshot() timed out waiting 30000ms to complete.`
I'm with cypress v12.17.3
I'm running the tests on github, with self-hosted runners and using node 18.16.0
Running the tests with parameter `--browser chrome`

**Comment by 0zzard** [comment]
Still got error `cy.screenshot() timed out waiting 30000ms to complete` after enabling `experimentalMemoryManagement: true` when running on azure dev ops. No issues when running on my local pc

**Comment by rhzs** [comment]
> Hi all, I first want to say I can understand the frustration with an issue like this. Especially when it seems simple, like there is a single solution. But from what I can tell, it's not just one error, but it's a condition Cypress reaches that could point back to multiple causes. That's one of the challenges with just providing a fix.
> 
> Anyway, a small update: it appears that Chrome's new headless mode that was turned on in [v12.15.0](https://docs.cypress.io/guides/references/changelog#12-15-0) is connected to at least some the more recent reports of problems above. That's suggested in this Discord [thread from today](https://discord.com/channels/755913899261296641/1138563361085333596/1138563361085333596), as well as the increased activity in these comments after 12.15.0 came out in June.
> 
> You can still opt-in to passing the old headless flag, as [described in our docs here](https://docs.cypress.io/api/plugins/browser-launch-api#Disable---headlessnew-for-Chrome), which should get you back to the default behavior prior to v12.15.0. If you started seeing this issue recently, and this is the reason, that should unblock you while we look into it further.
> 
> Also I am really curious if setting [`experimentalMemoryManagement`](https://docs.cypress.io/guides/references/experiments) to `true` in your config file might also make difference here. I may have missed it above, if anybody who experiences it regularly has already configured that, or if not, please let me know if changes the amount of times this happens.

I got the same error with screenshot timeout.
Does Cypress have a debugging log mechanism to debug the above issue? If you can suggest us, how to debug the actual problem, it'd be useful for us.

**Comment by theTestingApproach** [comment]
Tried with 12.17.4 with experimentalMemoryManagement: set to true and still have same issue

**Comment by SquadronROE** [comment]
I'm on 12.17.4, running tests in Browserstack. Locally the tests are great, but in Browserstack one of my specs dies with this error on a retry of a test. Wild.

I'll try to dig into that thread so we can continue debugging a bit more live. I think you're on to something Mark - it seems like it really matters when the underlying environment is starved of resources. So we might have to be a bit more careful with how we're putting together our tests to avoid unnecessary redirects and reloads, which are non-performant...

**Comment by theTestingApproach** [comment]
Hello,

Any resolution please? I see you guys have release 13.1.0 but i am stuck at 12.12.0

**Comment by alexanderalipoyo** [comment]
its already 2023 and still a problem

**Comment by AlexanderTunick** [comment]
Upgraded to Cypress 13.2.0 and now have dozens of auto-tests failing with this error. Interesting and frustrating actually. Though I even don't use cy.screenshot() anywhere

**Comment by marktnoonan** [comment]
@AlexanderTunick which version did you upgrade from? Does anything in this comment apply to you? https://github.com/cypress-io/cypress/issues/5016#issuecomment-1672439123

**Comment by AlexanderTunick** [comment]
@marktnoonan I've been upgrading sequentially from version to version (sometimes skipping several subversions) but the previous version was **13.1.0**, I will try your suggestion in https://github.com/cypress-io/cypress/issues/5016#issuecomment-1672439123 today and will be back with the result.

**Comment by AlexanderTunick** [comment]
@marktnoonan yes, can confirm that [experimentalMemoryManagement](https://docs.cypress.io/guides/references/experiments) set to `true` fixed this error.
(at least following the results of a single test cycle run)

cypress version **13.2.0**

**Comment by SaurabhBose** [comment]
Following up from my earlier message about `experimentalMemoryManagement: true `. We had another timeout which indicated the underlying cause was not cured fully. 

With `Upload screenshots and Videos on failure` in our github pipeline, we have now set `screenshotOnRunFailure: false` in cypress.config. My suggestion is to have the following to alleviate the pain. 

`experimentalMemoryManagement: true,
 screenshotOnRunFailure: false`

**Comment by AGroupman** [comment]
Also seeing this issue.

Just upgraded to Cypress 13.2.0, prior to the upgrade we were not having any issues with screenshots.
Since the upgrade we can no longer capture screenshots when running headless locally on Chrome.
Local headless screenshots appear to work fine when running on Electron, screenshots also seem to capture just fine when the tests are running in an azure devops pipeline (on Chrome).

Video capture appears to also be working fine in all cases.

**Comment by BarbaraGodayol** [comment]
I'm working with cypress 13.3.0 and I have the issue. After the test failed, we have the screenshot error. The test was executed on github pipeline: 
![image](https://github.com/cypress-io/cypress/assets/113983225/e4a2b3e1-5c6c-48f1-8a36-890418d9b65e)
It is randomly, not appears in all failed test

**Comment by barranquerox** [comment]
> I'm working with cypress 13.3.0 and I have the issue. After the test failed, we have the screenshot error. The test was executed on github pipeline: ![image](https://user-images.githubusercontent.com/113983225/271910483-e4a2b3e1-5c6c-48f1-8a36-890418d9b65e.png) It is randomly, not appears in all failed test

I have exactly the same issue @BarbaraGodayol

**Comment by dalechyn** [comment]
> I'm working with cypress 13.3.0 and I have the issue. After the test failed, we have the screenshot error. The test was executed on github pipeline: ![image](https://user-images.githubusercontent.com/113983225/271910483-e4a2b3e1-5c6c-48f1-8a36-890418d9b65e.png) It is randomly, not appears in all failed test

cypress 12.17.3, same issue here.

I guess that the github pipeline might have some issues

**Comment by bilarallen** [comment]
When running the test in CI using ver.13.3.0 and debian-linux a single test also fails, and upon checking in cypress-cloud test replay, it fails in the command `cy.screenshot()` although I don't have that command written in my test. When running locally using `npx cypress run...` this test is passing

![image](https://github.com/cypress-io/cypress/assets/65072341/4b2dda82-905d-46c6-a78c-893e50067df6)

```
cy.screenshot() timed out waiting 120000ms to complete.


View stack trace
at https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:134350:73 
at tryCatcher (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1807:23) 
at https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:4186:41 
at tryCatcher (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1807:23) 
at Promise._settlePromiseFromHandler (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1519:31) 
at Promise._settlePromise (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1576:18) 
at Promise._settlePromise0 (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1621:10) 
at Promise._settlePromises (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:1697:18) 
at _drainQueueStep (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:2407:12)
at _drainQueue (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:2400:9) 
at Async._drainQueues (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:2416:5) 
at Async.drainQueues (https://xxxxxxxxxxxxxxxxxxxxxx/__cypress/runner/cypress_runner.js:2286:14)
```

**Comment by DmitriiKrupinECM** [comment]
Not a solution, but a potential workaround to add to the commands.js file in order not to fail the hook (and just log the error) in case the screenshot fails to complete, is this command overwrite legit, @marktnoonan @jennifer-shehane ?
```
Cypress.Commands.overwrite('screenshot', (originalFn, name, options) => {
  try {
    originalFn(name, options)
  } catch (error) {
    Cypress.log({
      name: 'screenshot',
      message: [`Could not take a screenshot: ${error.message}`]
    })
  }
})
```

**Comment by bilarallen** [comment]
> Not a solution, but a potential workaround to add to the commands.js file in order not to fail the hook (and just log the error) in case the screenshot fails to complete, is this command overwrite legit, @marktnoonan @jennifer-shehane ?
> 
> ```
> Cypress.Commands.overwrite('screenshot', (originalFn, name, options) => {
>   try {
>     originalFn(name, options)
>   } catch (error) {
>     Cypress.log({
>       name: 'screenshot',
>       message: [`Could not take a screenshot: ${error.message}`]
>     })
>   }
> })
> ```

@DmitriiKrupinECM thanks but no luck with this workaround, already tried this on my end with test replay enabled in cypress cloud.

**Comment by DmitriiKrupinECM** [comment]
@bilarallen unfortunately I also started to receive the screenshot timeout even using this overwrite in combination with
`experimentalMemoryManagement: true` in configs.
And I'm not even using 13 yet (12.7)

**Comment by theTestingApproach** [comment]
Is it possible that this is being caused by the number of tests being run? I have 130+ tests and facing this issue from 12.14.0+ only in gitlab.. I very rarely run all the tests in GUI.

**Comment by g-zecchi** [comment]
> s it possible that this is being caused by the number of tests being run? I have 130+ tests and facing this issue from 12.14.0+ only in gitlab.. I very rarely run all the tests in GUI.

No, I'm facing this issue running only 13 tests.

**Comment by AlexanderTunick** [comment]
> @marktnoonan yes, can confirm that [experimentalMemoryManagement](https://docs.cypress.io/guides/references/experiments) set to `true` fixed this error. (at least following the results of a single test cycle run)
> 
> cypress version **13.2.0**

The fix did not last too long. Now receive this error again **13.2.0**

**Comment by jennifer-shehane** [comment]
We're taking another look into this. Spiking into it. We've looked into a few avenues in the past. It's still incredibly difficult to track down without a reproducible example that triggers it reliably. 

If anyone can share a reproducible example that we can run - and provide the info about where you're running it, how many CPU/resources you have, it'd be helpful.

It seems to be some kind of intermittent issue with the timing of things and perhaps available resources which makes this difficult to pin down.

**Comment by mschile** [comment]
Unfortunately, I have not been able to reproduce this issue yet nor is anything jumping out in the code as being incorrect as it seems to be [hanging/timing out](https://github.com/cypress-io/cypress/blob/257861bd6e3f7de77a4c7ede64e9538370c0a2ed/packages/driver/src/cy/commands/screenshot.ts#L105) when trying to capture the screenshot in [CDP](https://github.com/cypress-io/cypress/blob/257861bd6e3f7de77a4c7ede64e9538370c0a2ed/packages/server/lib/browsers/cdp_automation.ts#L484). I've tried running in a resource constrained docker container with a set of failing memory intensive tests. If someone who can reproduce the issue run with the following [debug](https://docs.cypress.io/guides/references/troubleshooting#Print-DEBUG-logs) logs that may help determine if memory is constrained when trying to capture the screenshot:

`DEBUG=cypress*memory*,cypress:server:screenshot,cypress-verbose:server:browsers:cri-client:*`

**Comment by rhzs** [comment]
@jennifer-shehane @mschile have you guys tried a "stress test" for capturing screenshot / videos? 
ie. capturing screenshot in parallel, etc.

**Comment by MadnezGirl** [comment]
same problem with cypress 12.17.3 when running 2-4 tests in a docker container locally

experimentalMemoryManagement: true - helped me

**Comment by jennifer-shehane** [comment]
@rhzs It's not recommended to run Cypress instances in parallel on the same machine - we don't fully support this usage. Is this what you're doing where you have the issue? There are a few other issues that can happen when this is done: https://github.com/cypress-io/cypress/issues/26211 https://github.com/cypress-io/cypress/issues/25470 https://github.com/cypress-io/cypress/issues/19565 There's a bunch more

**Comment by saas2813** [comment]
> We're taking another look into this. Spiking into it. We've looked into a few avenues in the past. It's still incredibly difficult to track down without a reproducible example that triggers it reliably.

Hi, When I was working with Cypress the problem showed up when a test failed on a page that was large enough to take long time to create the screenshot from, e.g. a listing from a database that failed to apply limits, thus the timeout was probably correct but since the reported error was the timeout not the actual assertion-failure that triggered the screenshot the reported failure did not give information.

Sorry, I do not have access to either the code or environment I was working on at the time.
It was quite reproducible, whenever there was a test-failure on a too large page I got the problem.
Since it is a timeout you would probably have an easier time reproducing it on a slow machine, but if you put a few MB of "lorem ipsum" in reasonably large font on a page and an assert-fail in a test for it you should be able to trigger the problem. (As I said a runaway DB-query will do it, and you really want to get "5M records returned when 20 was expected" instead of "Timeout" in that situation.) 
I should have mentioned I was running Headless, either locally or on our Jenkins CI.

**Comment by Sushmak0603** [comment]
Hi @jennifer-shehane 
Have we got any fix for this issue `cy.screenshot()` timed out waiting `30000ms` to complete. ??
I am currently using Cypress version 12.14.0
I see this issue only when I run the tests in the docker container and do not see this when I run it locally.

**Comment by Sushmak0603** [comment]
> Hi @jennifer-shehane Have we got any fix for this issue `cy.screenshot()` timed out waiting `30000ms` to complete. ?? I am currently using Cypress version 12.14.0 I see this issue only when I run the tests in the docker container and do not see this when I run it locally.

What worked for me is mentioning video: false, in cypress.config.js file in the module.exports section where e2e is folder containing my feature files.


module.exports = defineConfig({
  e2e: {
    video: false,

**Comment by AlexSdet** [comment]
Maybe this will be helpful. I found what was causing this issue on my side. One test was clicking link that opens new tab, and basically leave it hanging, and then next test was always failing with`cy.screenshot() timed out waiting 30000ms` .. as soon as I removed that link click - no more issues.

@jennifer-shehane I could be wrong, but I think the problem here is when Cypress loses focus on main browser tab or something of that nature.

**Comment by shiwantha-lakmal** [comment]
@AlexSdet  Agree with statment as per my investigation also this issue arise between multi state re-direction. 
this might not occure when doing element interaction within same context (page)

![image](https://github.com/cypress-io/cypress/assets/144337287/de49ba52-62db-4858-afc9-bc65846a1adb)

**Comment by jennifer-shehane** [comment]
@AlexSdet Yes, I agree this is certainly some of the cases where this happens. We still haven't had a reproducible example to show this timeout relaibly in this situation either though.

**Comment by Dale-777** [comment]
@jennifer-shehane we have this same issue with a single test, it occurs infrequently, and the reason it appears to be related to this test is the page under test in-particular is highly complex and no doubt is heavy to load & render, so until you have a way to simulate that you will not be able to reproduce this.

For us this error occurs in a feature file, with 8 scenarios, each loading the same page, so could be a memory issue.

```
cy.screenshot() timed out waiting 30000ms to complete.

Because this error occurred during a after each hook we are skipping all of the remaining tests.[Learn more](https://on.cypress.io/screenshot)
View stack trace
CypressError: `cy.screenshot()` timed out waiting `30000ms` to complete.

Because this error occurred during a `after each` hook we are skipping all of the remaining tests.
    at https://XXXX.com/__cypress/runner/cypress_runner.js:134337:73
    at tryCatcher (https://XXXX.com/__cypress/runner/cypress_runner.js:1807:23)
```

**Comment by Tathos20** [comment]
We are running multiple feature files in Azure pipelines. We see this issue always at the first feature file. All of the others are okay. Hope this information hepls.

**Comment by eschutho** [comment]
We're able to reproduce this consistently in Apache Superset (open source). Here's a [PR](https://github.com/apache/superset/pull/27095) that shows the E2E CI failures and the test output. The screenshot timeout errors seem to be limited to these code changes, too, but they pass when running Cypress locally. CI is running Chrome 119 on Ubuntu - 20.04.

**Comment by jennifer-shehane** [comment]
@eschutho Thanks for the repo! Any thoughts on updating to latest Cypress within those tests?

**Comment by jennifer-shehane** [comment]
@eschutho Could you give the minimal instructions to get this up and running with Cypress tests? I tried to follow the [contributing guidelines](https://github.com/apache/superset/blob/master/CONTRIBUTING.md#os-dependencies) and am still running into:

```
(venv)  superset db upgrade
zsh: command not found: superset
```

Also I get this error when running

```
pip install -e .
```

![Screenshot 2024-02-14 at 10 29 39 AM](https://github.com/cypress-io/cypress/assets/1271364/f304b513-f3cb-4cf8-80f3-bea8146523f9)
![Screenshot 2024-02-14 at 10 29 44 AM](https://github.com/cypress-io/cypress/assets/1271364/076adc76-b581-439f-878a-823226d9a517)

**Comment by eschutho** [comment]
> @eschutho Thanks for the repo! Any thoughts on updating to latest Cypress within those tests?

Hi, @jennifer-shehane yes, we're a little behind in our updates, but we are planning to upgrade this year. 

>Could you give the minimal instructions to get this up and running with Cypress tests?

You can use the make commands. `make superset` should install and build and `make flask-app` for backend `make node-app` for webpack. Once you have the app running, you can build/run cypress with:
`make build-cypress` and `make open-cypress 9000`

If you run into any issues with the make commands, it looks like with where you are currently, you'll need to run `pip install -r requirements/local.txt` and then follow the rest of the steps. 

Alternatively there's a docker image that you can use. These are the instructions for running with Docker.  https://github.com/apache/superset/blob/master/CONTRIBUTING.md?plain=1#L973

Happy to help with any other setup questions for reproducing.

**Comment by jennifer-shehane** [comment]
@eschutho make superset is failing and the docker image exits also. I'll need an easier example to run since this is now taking significant time to run. 
![Screenshot 2024-02-16 at 11 16 57 AM](https://github.com/cypress-io/cypress/assets/1271364/c0e2119c-6461-40e7-a2de-072644f59079)

**Comment by droca** [comment]
We were facing this issue as well, `only when running the tests on CI (Browserstack)` and `not all the time`. These are our conclusions:
 - The cause of this issue was that some of our tests where clicking on buttons and links that were `opening new tabs`. We didn't do anything on the new tabs and it was not having any effect on the e2e tests running locally.

 - The 'solution' we applied was to 'hack' a bit the behaviour of the application under test to avoid the links / buttons from opening a new tab. We did it with this code on the appropriate place:
 ```
 Cypress.$(element).on('click', e => {
   e.preventDefault()
})
 ```
 
 As a caveat, this solution only works for the regular elements / buttons that contain an href, but it doesn't with the ones `that are using the onClick function to redirect` (like with a react push or something like that)

**Comment by jennifer-shehane** [comment]
@droca I was able to recreate this situation. Only within the Chrome browser. This can be recreated with `cypress open` and `cypress run`.

`test.js`
```js
it('click on new tab', () => {
  cy.visit('index.html')
  cy.get('a').click().should('have.attr', 'foo')
})
```

`index.html`
```html
<html>
  <script>
  </script>
<body>
  <a target="_blank" href="new.html">Click me</a>
</body>
</html>
```

`new.html`
```html
<html>
</html>
```

![Screenshot 2024-02-26 at 11 15 38 AM](https://github.com/cypress-io/cypress/assets/1271364/310dd095-7baf-41aa-aeb8-20ad43b1e7b7)
